dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkflowInstanceServiceImpl.java

```diff
@@ -460,15 +470,45 @@ public Map<String, Object> queryTaskListByWorkflowInstanceId(User loginUser, lon
         List<TaskInstance> taskInstanceList =
                 taskInstanceDao.queryValidTaskListByWorkflowInstanceId(workflowInstanceId,
                         workflowInstance.getTestFlag());
+        List<TaskInstanceDependentDetails> taskInstanceDependentDetailsList =
+                setTaskInstanceDependentResult(taskInstanceList);
+
         Map<String, Object> resultMap = new HashMap<>();
         resultMap.put(WORKFLOW_INSTANCE_STATE, workflowInstance.getState().toString());
-        resultMap.put(TASK_LIST, taskInstanceList);
+        resultMap.put(TASK_LIST, taskInstanceDependentDetailsList);
         result.put(DATA_LIST, resultMap);
 
         putMsg(result, Status.SUCCESS);
         return result;
     }
 
+    private List<TaskInstanceDependentDetails> setTaskInstanceDependentResult(List<TaskInstance> taskInstanceList) {
```

Comment:
```suggestion
    private List<TaskInstance> setTaskInstanceDependentResult(List<TaskInstance> taskInstanceList) {
```
We should return `List<TaskInstance>` rather than `List<TaskInstanceDependentDetails>`

Reply:
Fixed.